pkg/daemon/ceph/osd/nsenter_test.go

```diff
@@ -38,12 +38,22 @@ func TestCheckIfBinaryExistsOnHost(t *testing.T) {
 	executor := &exectest.MockExecutor{}
 	executor.MockExecuteCommandWithCombinedOutput = func(command string, args ...string) (string, error) {
 		logger.Infof("%s %v", command, args)
-		if command == "nsenter" && args[0] == "--mount=/rootfs/proc/1/ns/mnt" && args[1] == "--" && args[3] == "help" {
-			if args[2] == "/usr/sbin/lvm" || args[2] == "/sbin/lvm" {
-				return "success", nil
+		switch command {
+
+		case "nsenter":
+			if args[0] == "--mount=/rootfs/proc/1/ns/mnt" && args[1] == "--" && args[3] == "help" {
+				if args[2] == "/usr/sbin/lvm" || args[2] == "/sbin/lvm" {
+					return "success", nil
+				}
 			}
-		}
 
+		case "/rootfs/usr/sbin/lvm":
+			return "success", nil
```

Comment:
To test this route, the test code should be like the following.

```
        // nsenter exists here
	executor.MockExecuteCommandWithCombinedOutput = func(command string, args ...string) (string, error) {
		logger.Infof("%s %v", command, args)
		if command == "nsenter" && args[0] == "--mount=/rootfs/proc/1/ns/mnt" && args[1] == "--" && args[3] == "help" {
			if args[2] == "/usr/sbin/lvm" || args[2] == "/sbin/lvm" {
				return "success", nil
                }
		return "", errors.Errorf("unknown command %s %s", command, args)
        }
	context := &clusterd.Context{Executor: executor}
	ne := NewNsenter(context, lvmCommandToCheck, []string{"help"})
	err := ne.checkIfBinaryExistsOnHost()
	assert.NoError(t, err)

	executor.MockExecuteCommandWithCombinedOutput = func(command string, args ...string) (string, error) {
		logger.Infof("%s %v", command, args)
		switch command {
                case "nsenter":
                        return "", errors.New("failed to call nsenter")

		case "/rootfs/usr/sbin/lvm":
			return "success", nil

                }
		return "", errors.Errorf("unknown command %s %s", command, args)
	}
	context := &clusterd.Context{Executor: executor}
	ne := NewNsenter(context, lvmCommandToCheck, []string{"help"})
	err := ne.checkIfBinaryExistsOnHost()
	assert.NoError(t, err)
```

Reply:
Yes good point otherwise we always fall into the first one.